File: encryption_example/Encryption.php
<?php

function get_signed_params($public_key, $secret_key, $params) {
    $params = array_merge(['api_key' => $public_key], $params);
    ksort($params);
    //decode return value of http_build_query to make sure signing by plain parameter string
    $signature = hash_hmac('sha256', urldecode(http_build_query($params)), $secret_key);
    return http_build_query($params) . "&sign=$signature";
}

$params = [
	'symbol' => 'BTCUSDT', 
	'side' => 'Buy', 
	'order_type' => 'Limit', 
	'qty' => '0.01', 
	'price' => '5000', 
	'time_in_force' => 'GoodTillCancel',
	'reduce_only' => false,
	'close_on_trigger' => false,
	'timestamp' => time() * 1000
];

$url = 'https://api.bybit.com/private/linear/order/create';

$public_key = 'Un4rOHOZ2VpQhheMJ';
$secret_key = 'SbmtjCvFNFXx8Km5DY1hx6WxIWus5B6Iheh';
$qs=get_signed_params($public_key, $secret_key, $params);
$curl_url=$url."?".$qs;
$curl=curl_init($curl_url);
echo $curl_url;
curl_setopt($curl, CURLOPT_URL, $curl_url);
#curl_setopt($curl, CURLOPT_POSTFIELDS, $qs);
curl_setopt($curl, CURLOPT_POST, true);
curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($curl, CURLOPT_FOLLOWLOCATION, 1);
curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, 0);
curl_setopt($curl, CURLOPT_SSL_VERIFYHOST, 0);
#curl_setopt($curl, CURLOPT_PROXY,"127.0.0.1:1087");
$response=curl_exec($curl);
echo $response;

File: encryption_example/Encryption.py
import hashlib
import hmac
import json
import requests
import urllib3
import time
from urllib.parse import quote_plus


def create(api_key, secret_key, side, order_type, qty, price):
    params = {  # futures create order params
        "side": side,
        "symbol": "BTCUSD",
        "order_type": order_type,
        "qty": qty,
        "price": price,
        "time_in_force": "GoodTillCancel",
        "api_key": api_key,
        "timestamp": round(time.time() * 1000),
        "recv_window": 10000,
    }
    '''
    params = {  # spot create order params
        "side": side,
        "symbol": "BTCUSDT",
        "type": order_type,
        "qty": qty,
        "price": price,
        "timeInForce": "GTC",
        "api_key": api_key,
        "timestamp": round(time.time() * 1000),
        "recv_window": 10000,
    }
    '''
    '''
    params = {  # /v2/private/order/list example demonstrating correct encoding
        "symbol": "BTCUSD",
        "cursor": "42T+FpoQ6orO6YfV/coAZs/pGP2dMKaju56oRjWCOevTN6g5vDf8xFia5iyJYY3o",
        "api_key": api_key,
        "timestamp": round(time.time() * 1000),
        "recv_window": 10000,
    }
    '''
    # Create the param str
    param_str = ""
    for key in sorted(params.keys()):
        v = params[key]
        if isinstance(params[key], bool):
            if params[key]:
                v = "true"
            else:
                v = "false"
        param_str += f"{key}={v}&"
    param_str = param_str[:-1]
    print(param_str)

    # Generate the signature
    hash = hmac.new(secret_key, param_str.encode("utf-8"), hashlib.sha256)
    signature = hash.hexdigest()
    sign_real = {
        "sign": signature
    }

    # Prepare params in the query string format
    # quote_plus helps quote rare characters like "/" and "+"; this must be
    # applied after the signature generation.
    param_str = quote_plus(param_str, safe="=&")
    full_param_str = f"{param_str}&sign={sign_real['sign']}"
    print(f"full_param_str = {full_param_str}")

    # Request information
    method = "POST"
    url = "https://api-testnet.bybit.com/v2/private/order/create"
    #url = "https://api-testnet.bybit.com/spot/v1/order"
    #url = "https://api-testnet.bybit.com/v2/private/order/list"
    if "spot" in url or method == "GET":
        headers = {"Content-Type": "application/x-www-form-urlencoded"}
        body = None
    else:
        headers = {"Content-Type": "application/json"}
        body = dict(params, **sign_real)

    urllib3.disable_warnings()
    s = requests.session()
    s.keep_alive = False

    # Send the request(s)
    for x in range(1):
        if "spot" in url:
            # Send a request to the spot API
            response = requests.request(method, url, data=full_param_str,
                                        headers=headers, verify=True)
            #response = requests.post(f"{url}?{full_param_str}",
            #                         headers=headers, verify=False)
        else:
            # Send a request to the futures API
            if method == "POST":
                response = requests.request(method, url, data=json.dumps(body),
                                            headers=headers, verify=False)
            else:  # GET
                response = requests.request(method, f"{url}?{full_param_str}",
                                            headers=headers, verify=False)
        print(response)
        print(response.text)


def main():
    api_key = ""
    secret_key = b""
    create(api_key, secret_key, "Buy", "Limit", 1, 8700)


if __name__ == "__main__":
    main()
